File: src/main/java/bad/robot/radiate/teamcity/BuildType.java
package bad.robot.radiate.teamcity;

import bad.robot.radiate.Hypermedia;

class BuildType extends TeamCityObject implements Hypermedia {

    private final String id;
    private final String name;
    private final String href;
    private final String projectName;
    private final String projectId;

    public BuildType(String id, String name, String href, String projectName, String projectId) {
        this.id = id;
        this.name = name;
        this.href = href;
        this.projectName = projectName;
        this.projectId = projectId;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    @Override
    public String getHref() {
        return href;
    }

    public String getProjectName() {
        return projectName;
    }

    public String getProjectId() {
        return projectId;
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/BuildType.scala
package bad.robot.radiate.teamcity

import bad.robot.radiate.Hypermedia
import argonaut._
import Argonaut._

object BuildType {
  implicit def buildTypeCodec: DecodeJson[BuildType] = {
    DecodeJson(cursor => {
      for {
        id <- (cursor --\ "id").as[String]
        name <- (cursor --\ "name").as[String]
        href <- (cursor --\ "href").as[String]
        projectName <- (cursor --\ "projectName").as[String]
        projectId <- (cursor --\ "projectId").as[String]
      } yield {
        BuildType(id, name, href, projectName, projectId)
      }
    })
  }
}

case class BuildType(id: String, name: String, href: String, projectName: String, projectId: String) extends TeamCityObject with Hypermedia
